executor/src/witgen/jit/variable.rs

```diff
@@ -75,10 +76,12 @@ impl Variable {
 }
 
 #[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Debug)]
-pub struct MachineCallVariable {
+pub struct MachineCallVariable<T> {
     pub identity_id: u64,
     pub row_offset: i32,
     pub index: usize,
+    // TODO: Remove this after we changed identity_id: u64 to bus_id: T
+    pub _phantom: PhantomData<T>,
```

Comment:
This is the main change, all other changes are just consequences of this. See [this change](https://github.com/powdr-labs/powdr/pull/2488/files#diff-e88dd6f458f95eb3713fd4d0f2cc7fe3a77c966d13f821ae09d52635022818ebR79) in #2488.